api/institutions/authentication.py

```diff
@@ -0,0 +1,68 @@
+import json
+
+import jwe
+import jwt
+
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.authentication import BaseAuthentication
+
+from api.base import settings
+from website.project.model import Institution
+from framework.auth import get_or_create_user
+
+
+class InstitutionAuthentication(BaseAuthentication):
+    media_type = 'text/plain'
+
+    def authenticate(self, request):
+        try:
+            payload = jwt.decode(
+                jwe.decrypt(request.body, settings.JWE_SECRET),
+                settings.JWT_SECRET,
+                options={'verify_exp': False},
+                algorithm='HS256'
+            )
+        except (jwt.InvalidTokenError, KeyError, ValueError, TypeError):
```

Comment:
You are handling many errors the same way. At the very least, it would be good to have different error messages for different kinds of errors; this will help with debugging. Also, I'm not clear on why all these errors need to be caught. `InvalidTokenError` makes sense, but why `KeyError`, `ValueError` and `TypeError`?